sw/dnn/layernorm/src/layernorm.h

```diff
@@ -294,11 +765,25 @@ static inline void layernorm_layer(layernorm_layer_t l) {
         // Compute layernorm tile
         if (snrt_is_compute_core()) snrt_mcycle();
         if (l.baseline == 1) {
-            layernorm_fp32(local_itile, local_otile, l.batch_size, tile_seq_len,
-                           l.embeddings, l.eps);
+            if (l.dtype == FP32)
+                layernorm_fp32_naive(local_itile, local_otile, l.batch_size,
+                                     tile_seq_len, l.embeddings, l.eps);
+            else if (l.dtype == FP16)
+                layernorm_fp16_naive(local_itile, local_otile, l.batch_size,
+                                     tile_seq_len, l.embeddings, l.eps);
         } else {
-            layernorm_fp32_opt(local_itile, local_otile, l.batch_size,
-                               tile_seq_len, l.embeddings, l.eps);
+            if (l.dtype == FP32) {
+                layernorm_fp32_opt(local_itile, local_otile, l.batch_size,
+                                   tile_seq_len, l.embeddings, l.eps);
+            } else if (l.dtype == FP16) {
+                layernorm_fp16_opt(local_itile, local_otile, l.batch_size,
+                                   tile_seq_len, l.embeddings, l.eps);
+            } else if (l.dtype == FP8) {
+                layernorm_fp8_opt(local_itile, local_otile, l.batch_size,
+                                  tile_seq_len, l.embeddings, l.eps);
+            } else {
+                printf("Unsupported data type\n");
+            }
         }
```

Comment:
For consistency with the GEMM implementation I would first test dtype with a concise switch case statement, and internally for each precision select between the different implementations